strutil/pathiter.go

```diff
@@ -64,13 +64,13 @@ func (iter *PathIterator) Path() string {
 }
 
 // CurrentName returns the name of the current path element.
-// The return value may end with '/'. Use CleanName to avoid that.
+// The return value may end with '/'. Use CurrentNameNoSlash to avoid that.
```

Comment:
PathInterator could use definitely some more renamings/cleanups but is fine for here I suppose

Reply:
Agreed. Created maintenance card: https://warthogs.atlassian.net/browse/SNAPDENG-10719